src/averageAges.js

```diff
@@ -37,7 +47,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womenInCentury = {};
+
+  if (withChildren) {
+    womenInCentury = people.filter(
+      person => person.sex === 'f' && people.some(
+        children => children.mother === person.name));
+  } else {
+    womenInCentury = people.filter(
+      person => person.sex === 'f');
+  }
+
+  const sumOfAges = womenInCentury.reduce(
+    (sum, person) => sum + person.died - person.born, 0);
+
+  return sumOfAges / womenInCentury.length;
```

Comment:
the same code you have for calculateMenAverageAge, so you need to create separated function like `calculateAvarage() `and reuse as for calculateMenAverageAge and for calculateWomenAverageAge()